e2e/test/iothub/service/RegistryManagerE2ETests.cs

```diff
@@ -87,6 +88,62 @@ public async Task RegistryManager_AddDeviceWithTwinWithDeviceCapabilities()
             }
         }
 
+        [LoggedTestMethod]
+        public async Task RegistryManager_BulkLifecycle()
+        {
+            int bulkCount = 50;
+            string[] deviceIds = new string[bulkCount];
+            List<Device> devices = new List<Device>();
+            for (int i = 0; i < bulkCount; i++)
+            {
+                deviceIds[i] = _devicePrefix + Guid.NewGuid();
+                devices.Add(new Device(deviceIds[i]));
+            }
+
+            var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            // Test that you can create devices in bulk
+            var bulkAddResult = await registryManager.AddDevices2Async(devices).ConfigureAwait(false);
+            Assert.IsTrue(bulkAddResult.IsSuccessful);
+
+            foreach (Device device in devices)
+            {
+                // After a bulk add, every device should be able to be retrieved
+                Assert.IsNotNull(await registryManager.GetDeviceAsync(device.Id).ConfigureAwait(false));
+            }
+
+            List<Twin> twins = new List<Twin>();
+            string expectedProperty = "someNewProperty";
+            string expectedPropertyValue = "someNewPropertyValue";
+            foreach (string deviceId in deviceIds)
+            {
+                Twin twin = await registryManager.GetTwinAsync(deviceId).ConfigureAwait(false);
+                twin.Properties.Desired[expectedProperty] = expectedPropertyValue;
+                twins.Add(twin);
+            }
+
+            // Test that you can update twins in bulk
+            await registryManager.UpdateTwins2Async(twins).ConfigureAwait(false);
+
+            foreach (string deviceId in deviceIds)
+            {
+                Twin twin = await registryManager.GetTwinAsync(deviceId).ConfigureAwait(false);
+                Assert.IsNotNull(twin.Properties.Desired[expectedProperty]);
+                Assert.AreEqual(expectedPropertyValue, (string)twin.Properties.Desired[expectedProperty]);
+            }
+
+            // Test that you can delete device identities in bulk
+            var bulkDeleteResult = await registryManager.RemoveDevices2Async(devices, true, default).ConfigureAwait(false);
+
+            Assert.IsTrue(bulkDeleteResult.IsSuccessful);
+
+            foreach (Device device in devices)
+            {
+                // After a bulk delete, every device should not be found
+                Assert.IsNull(await registryManager.GetDeviceAsync(device.Id).ConfigureAwait(false));
```

Comment:
Will this not throw 404?

Reply:
Surprisingly, it doesn't. I was expecting it to, but it just returns null

Reply:
I'm not looking to change the behavior of the API at this point, though

Reply:
Ohh really? That is surprising to me :) Ofcourse we cannot change the behavior of the API but I was expecting a 404.